app/controllers/api/v1/items_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Api::V1::ItemsController < ApplicationController
+  before_action :set_item, only: %i[show update destroy]
+
+  # GET /api/v1/items
+  def index
+    @items = Item.all.includes(:reservations)
+
+    render json: @items
+  end
+
+  # GET /api/v1/items/1
+  def show
+    render json: @item
+  end
+
+  # POST /api/v1/items
+  def create
+    @item = Item.new(item_params)
+
+    if @item.save
+      render json: @item, status: :created, location: api_v1_item_url(@item)
+    else
+      render json: @item.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /api/v1/items/1
+  def update
+    if @item.update(item_params)
+      render json: @item
+    else
+      render json: @item.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /api/v1/items/1
+  def destroy
+    @item.destroy
+    render json: { message: 'Item deleted successfully.' }
+  end
```

Comment:
- [ ] What happens if this action fails?  Kindly, use some error handling in this action. 👍

Reply:
OK, I will add actions to show fails